### PR TITLE
Bump github.com/gardener/gardener to 1.86.0

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area open-source
/kind enhancement
**What this PR does / why we need it**:
This PR updates the `github.com/gardener/gardener` dependency to `v1.86.0`. As a consequence, admission CA and service certificates are now managed automatically. The vendor directory is removed as well.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Bump github.com/gardener/gardener to 1.86.0.
```
```breaking operator
CA and server certificates for the admission component are managed automatically. Passing custom certificates via Helm values is not supported anymore.
```
```other developer
The `vendor` directory was removed in favor of the `go mod cache`.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,7 +45,6 @@ include vendor/github.com/gardener/gardener/hack/tools.mk
 .PHONY: start
 start:
 	@LEADER_ELECTION_NAMESPACE=garden GO111MODULE=on go run \
```
Comment (ScheererJ) [inline Makefile:57]: Do we still need `GO111MODULE=on`?

@MartinWeindel removed those entries in similar pull requests for other extensions, e.g. https://github.com/gardener/gardener-extension-shoot-networking-filter/pull/111.

File: Makefile [hunk 2]
```diff
@@ -83,49 +83,47 @@ docker-images:
...
-revendor:
+.PHONY: tidy
+tidy:
 	@GO111MODULE=on go mod tidy
-	@GO111MODULE=on go mod vendor
-	@chmod +x $(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/*
-	@chmod +x $(REPO_ROOT)/vendor/github.com/gardener/gardener/hack/.ci/*
-	@$(REPO_ROOT)/hack/update-github-templates.sh
-	@ln -sf ../vendor/github.com/gardener/gardener/hack/cherry-pick-pull.sh $(HACK_DIR)/cherry-pick-pull.sh
+	@mkdir -p $(REPO_ROOT)/.ci/hack && cp $(GARDENER_HACK_DIR)/.ci/* $(REPO_ROOT)/.ci/hack/ && chmod +xw $(REPO_ROOT)/.ci/hack/*
```
Comment (ScheererJ) [inline Makefile:101]: Do we not need to copy `$(GARDENER_HACK_DIR)/generate-controller-registration.sh` as well similar to https://github.com/gardener/gardener-extension-shoot-networking-filter/pull/111?

File: hack/update-codegen.sh [hunk 3]
```diff
@@ -26,36 +31,36 @@ rm -f $GOPATH/bin/*-gen
 
 PROJECT_ROOT=$(dirname $0)/..
```
Comment (ScheererJ) [inline hack/update-codegen.sh:32]: `PROJECT_ROOT` seems to be not used anymore. Can we remove it?

**Review by ScheererJ — CHANGES_REQUESTED** [review]
Thanks for this huge change. I have a few questions.

**Comment by ScheererJ** [comment]
Can we remove https://github.com/gardener/gardener-extension-networking-calico/blob/2ab298f479ca9a637caa74317e1101d7a3f99b50/.ci/verify#L12C1-L15C68 now similar to https://github.com/gardener/gardener-extension-shoot-networking-filter/pull/111?

File: Makefile (hunk 2 above)
Comment (timuthy) [reply Makefile:101]: It's at least not added to the AWS extension ([ref](https://github.com/gardener/gardener-extension-provider-aws/blob/d75666468dda0fe76ff131d91ca24921dccfa578/Makefile#L106)) and I don't see a reason why this was done. Maybe @MartinWeindel can tell?

**Comment by ScheererJ** [comment]
Can you please adapt https://github.com/gardener/gardener-extension-networking-calico/blob/master/.github/workflows/vendor_gardener.yaml as well (revendor => tidy) similar to https://github.com/gardener/gardener-extension-shoot-networking-filter/pull/111?

File: Makefile (hunk 2 above)
Comment (ScheererJ) [reply Makefile:101]: Checked offline with @MartinWeindel. From his point of view, this should work. Hence, I am fine with it.

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Pull request merged**